pkg/infra/iac2/templates_compiler.go

```diff
@@ -192,61 +200,79 @@ func (tc TemplatesCompiler) renderResource(out io.Writer, resource core.Resource
 	}
 	inputArgs := make(map[string]templateValue)
 	for fieldName := range tmpl.InputTypes {
-		// dependsOn will be a reserved field for us to use to map dependencies. If specified as an Arg we will automatically call resolveDependencies
-		if fieldName == "dependsOn" {
-			inputArgs[fieldName] = stringTemplateValue{value: tc.resolveDependencies(resource)}
-			continue
-		}
-		if fieldName == "protect" {
-			inputArgs[fieldName] = stringTemplateValue{value: "protect", raw: "protect"}
-			continue
-		}
-		childVal := resourceVal.FieldByName(fieldName)
-		structField, found := resourceVal.Type().FieldByName(fieldName)
-		iacTag := ""
-		if found {
-			iacTag = structField.Tag.Get("render")
-		}
-
-		var err error
-		var resolvedValue templateValue
-		if iacTag == "template" {
-			resolvedValue = nestedTemplateValue{
-				parentValue:            resourceVal,
-				childValue:             childVal,
-				iacTag:                 iacTag,
-				tc:                     &tc,
-				useDoubleQuotedStrings: false,
+		func(fieldName string) {
+			defer func() {
+				r := recover()
+				if r == nil {
+					return
+				}
+				panic(errors.Errorf("panic rendering field %s: %v", fieldName, r))
+			}()
+			// dependsOn will be a reserved field for us to use to map dependencies. If specified as an Arg we will automatically call resolveDependencies
+			if fieldName == "dependsOn" {
+				inputArgs[fieldName] = stringTemplateValue{value: tc.resolveDependencies(resource)}
+				return
 			}
```

Comment:
why make this change? Also wont this cause depends on and protect to be handled differently when they are both our reserved names and should be handled in the same manner?

Reply:
The only thing this changed was to put it into a function with the `defer func()` to wrap panics with the relevant field name and resource Id. All the reflection stuff likes to panic, and the resulting message makes it hard to diagnose where the problem was. This is a band-aid solution (there's more robust / cleaner solutions), but good enough for now

Reply:
i see, why is depends on in here instead of the other portion or why isnt protect in here as well since thats also a reserved field for us

Reply:
The code is the same as before except instead of `continue` it's `return`. Anything else is likely a merge/rebase error, I'll take another look at it.

Reply:
I don't think we need a full try/catch - these can be resolved by properly `Kind` checking or `IsNil` checking in various places. I didn't want to go through the current code with a fine-toothed comb at this point though. I think it originally came up when looking into https://github.com/klothoplatform/klotho/issues/476 (but I dropped the investigation and just commented out the expose and let Jordan fix it).

Reply:
But you have implemented a full try-catch, just in a kinda weird way (and specifically, by doing it as a catch-try, which is what Go forced you to do).

Reply:
I meant a generalized try/catch. The idiomatic Go thing would be to split `func(fieldName string)` into a real function and call it from within the loop, which can then more easily catch and handle any panics. Again, there's much better/cleaner ways to do this - and nothing panics right now so we don't _need_ it, but I figured it was better than nothing. I can revert this part if it's distracting from the other parts of the PR.

Reply:
fwiw, I got distracted (😬) and looked into what a nice try-catch API might look like, cause it piqued my interest. This is what I came up with: https://gist.github.com/yuval-klotho/c8788725a8967b33464b4b7489e55cae